### ISSUE TITLE
All entities regularly, but randomly, become 'unavailable' for a significant period of time

### ISSUE THREAD

**Issue by ChrisJ60** [megakid/ha_octopus_intelligent#52]
For some while now (I only just became aware of this after an incident last night) all of the entities provided by thsi integration become unavailable for significant periods of time. This occurs seemingly randomly.  My Internet connection is rock solid so it isn't that. I'm not having any issues with any other integrations in general nor with the 'Octopus Energy' integration specifically so it seems to be a problem with thsi specific integration.

Is anyone else experiencing this?

**Comment by stubbo66** [comment]
I only started setting up my integration a few days ago and it has been an almost daily occurrence.

It also seems to follow a pattern of being up and then down for 10-20 minutes at a time over a 6 hour period, then stabilises again. Until today it had mostly been late evening / early morning, but today it has been up and down since about 9am.

I can only assume they are working on the APIs on the Octopus end, but I can't find any sort of API availability information from their website.

**Comment by ChrisJ60** [comment]
I'd be surprised if the issue is at the Octopus end since the 'Octopus Energy' integration, which uses the same APIs is super stable. I'm kind of at a loss as too what the issue might be.

**Comment by monkga1** [comment]
Same and it's been like this for a long time

**Comment by godfreym29** [comment]
Try reverting to [v1.7.0](https://github.com/megakid/ha_octopus_intelligent/releases/tag/v1.7.0).
It seems to have solved my problems for now.

**Comment by stubbo66** [comment]
1.7.0 doesn’t appear to have made any difference to me, still not staying active

**Comment by godfreym29** [comment]
> 1.7.0 doesn’t appear to have made any difference to me, still not staying active

After not working at all for a couple of days, it fixed mine for the afternoon, but has since reverted to missing data, but every so often at least it is retrieving something ...

**Comment by sandyman44** [comment]
got the same thing myself this morning. happened to be watching it as I was checking my automation to change powerwall backup reserve.

I restarted HA with debug logging enabled on this integration, got the attached, hopefully this can help someone analyse the issue?  I've obfuscated the token values.
[iointegrationerror.txt](https://github.com/user-attachments/files/17686924/iointegrationerror.txt)

**Comment by thestumonkey** [comment]
I just tried updating to the latest HA version and removing my config and adding back.  It won't even setup and add the entities now.  "Unexpected error"

**Comment by thestumonkey** [comment]
I'm getting the same massive error message.  I'm guessing that it's sending something that is deprecated, which is why the response is a list of all the fields?

**Comment by sandyman44** [comment]
I just switched to the bottlecapdave integration. worked first time.

**Comment by ChrisJ60** [comment]
I strongly recommend switching to the bottlecapdave integration. That is being actively maintained/enhanced and seems to work very well.

**Comment by monkga1** [comment]
However I think that the other integration does not have the sensors to detect longer additional slots to turn in things like washing machines & dishwashers.

Regards
Gav
________________________________

**Comment by ChrisJ60** [comment]
The sensor binary_sensor.octopus_energy_a_**_YourAccountNumber_**_intelligent_dispatching is 'on' whenever IOG is giving you cheap rate electricity, with during the core 23:30 - 05:30 period but at any other time to. I have automations that use this to e.g. prevent my home batteries discharging if the car happens to charge outside of the core period.

**Comment by whistlebare** [comment]
Same issues for me too...it's driving me crazy.  Will take a look at this bittlecapdave integration

**Comment by stubbo66** [comment]
I've installed and changed all my automations to use the bottlecapdave integration now. That does appear to be much more stable.

All of the data is there, some if it is just harder to get at.

Thankfully for me the only real thing I need is the intelligent dispatching  sensor so I can switch on or off the charging of my in home batteries if they need a top up, or to prevent my car from draining the batteries down.

Actually freeing myself from the start and end time of sessions has made the scripts a lot easier.

**Comment by thestumonkey** [comment]
As mentioned above, the binary_sensor.intelligent dispatching will tell you
if cheap rate or not.
In addition it has attrs which tell you the dispatching slots as well as
the static off peak ones

**Comment by pwwukpw** [comment]
Same issue here

**Comment by nivek1612** [comment]
I ditched it and only use BottlecapDaves integration now. This integration has not been touched for a while

**Comment by CaptainHam** [comment]
> I am also getting this unavailable now, I am wondering if octopus have changed its api, and this integration may not get updated now?
> 

I noticed my integration was still stuck in v1, so i deleted it and reinstalled from HACS and it's now on the correct version of 1.9. The entities seem to be working again now

**Comment by f1wade** [comment]
My issue was related to octopus api being down.
Working now

**Comment by CaptainHam** [comment]
> My issue was related to octopus api being down. Working now

Same here but still required uninstall/reinstall before it worked again, so just commenting in case it helps anyone else

**Comment by whistlebare** [comment]
Yes just ho e from work and can too confirm re-add of account details and api key has brought them back to like   
  

  

  

  
  
  
  
  
>
